hazelcast/src/main/java/com/hazelcast/instance/DelegatingAddressPicker.java

```diff
@@ -0,0 +1,99 @@
+package com.hazelcast.instance;
+
+import com.hazelcast.config.NetworkConfig;
+import com.hazelcast.nio.Address;
+import com.hazelcast.nio.IOUtil;
+import com.hazelcast.spi.AddressLocator;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.nio.channels.ServerSocketChannel;
+import java.util.concurrent.TimeUnit;
+
+
+public class DelegatingAddressPicker implements AddressPicker {
```

Comment:
Suggest to add unit tests for it.

Reply:
sure thing, working on them right now. together with declarative configurations